src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -59,6 +59,32 @@ static bool check_vbox(const TypeInstPtr* vbox_type) {
 }
 #endif
 
+bool LibraryCallKit::arch_supports_vector_rotate(int opc, int num_elem, BasicType elem_bt, bool check_bcast) {
+    bool is_supported = true;
+    if (!arch_supports_vector(opc, num_elem, elem_bt, VecMaskNotUsed, true /*has_scalar_args*/) ||
+        (check_bcast && !arch_supports_vector(VectorNode::replicate_opcode(elem_bt), num_elem, elem_bt, VecMaskNotUsed))) {
+      is_supported = false;
+    }
+    int lshiftopc = VectorNode::opcode(elem_bt == T_LONG ? Op_LShiftL : Op_LShiftI, elem_bt);
+    auto urshiftopc = [&]() {
+      switch(elem_bt) {
+        case T_INT: return Op_URShiftI;
+        case T_LONG: return Op_URShiftL;
+        case T_BYTE: return Op_URShiftB;
+        case T_SHORT: return Op_URShiftS;
+        default: return (Opcodes)0;
+      }
+    };
+    int rshiftopc = VectorNode::opcode(urshiftopc(), elem_bt);
+    if (!is_supported &&
+        arch_supports_vector(lshiftopc, num_elem, elem_bt, VecMaskNotUsed) &&
+        arch_supports_vector(rshiftopc, num_elem, elem_bt, VecMaskNotUsed) &&
+        arch_supports_vector(Op_OrV, num_elem, elem_bt, VecMaskNotUsed)) {
+      is_supported = true;
+    }
```

Comment:
check_bcast is true only when shift value is a non-constant scalar value, in that case we need to check for broadcasting operation for shift, in all other cases broadcast is not needed.  Constant shift value is an optimizing case since AVX512 offers instructions which directly accept 8bit immediate shift value.